### ISSUE TITLE
DirectML + ssd_mobilenet_v1_10.onnx Issue

### ISSUE THREAD

**Issue by vincent911001** [onnx/models#438]
Hi,

I have tried to create a gstreamer plugin that uses ONNX-DirectML to run [ssd_mobilenet_v1_10.onnx](https://github.com/onnx/models/tree/master/vision/object_detection_segmentation/ssd-mobilenetv1/model) from ONNX model zoo.

FYI, I have successfully replicated the python sample to run ssd_mobilenet_v1_10.onnx with DirectML in a C++ sample app and I am getting the same results by using the same test image.

The issue happens when i was trying to integrate the codes to make a gstreamer plugin.
Below is the code snippet of the implementation:
```
void DMLDetect::Init()
{
	_env = std::make_unique<Ort::Env>(ORT_LOGGING_LEVEL_WARNING, _LOGGING_ID.c_str());

	auto providers = Ort::GetAvailableProviders();
	for (auto provider : providers)
		std::cout << provider << std::endl;

	_session_options.DisableMemPattern();
	_session_options.SetExecutionMode(ORT_SEQUENTIAL);
	_session_options.SetIntraOpNumThreads(1);
	// set provider
	OrtSessionOptionsAppendExecutionProvider_DML(_session_options, 0);
	_session_options.SetGraphOptimizationLevel(GraphOptimizationLevel::ORT_ENABLE_EXTENDED);

	std::cout << "Using Onnxruntime C++ API" << std::endl;
	_memory_info = std::make_unique<Ort::MemoryInfo>(Ort::MemoryInfo::CreateCpu(OrtArenaAllocator, OrtMemTypeDefault));
}

/**
* create input tensor with input video frame
* expect img in RGB format
**/
void DMLDetect::SetInputFrame(cv::Mat & img)
{
	unsigned const img_width = img.cols;
	unsigned const img_height = img.rows;
	unsigned const img_channel = img.channels();

	_input_node_dims.resize(_INPUT_NODE_DIM_SIZE);
	// NHWC order
	// N = batch size = 1 in this case
	_input_node_dims[0] = 1;
	_input_node_dims[1] = img_height;
	_input_node_dims[2] = img_width;
	_input_node_dims[3] = img_channel;
	_input_tensor_size = img_width * img_height * img_channel;

	// create input tensors
	_input_tensors.push_back(Ort::Value::CreateTensor<uint8_t>(*_memory_info, img.data, _input_tensor_size, _input_node_dims.data(), _INPUT_NODE_DIM_SIZE));
}

/**
* perform inference with the input frame
**/
void DMLDetect::Infer()
{
	_output_tensors = _session->Run(Ort::RunOptions{ nullptr }, _INPUT_NODE_NAMES.data(), _input_tensors.data(), _INPUT_NODE_NAMES_SIZE, 
		_OUTPUT_NODE_NAMES.data(), _OUTPUT_NODE_NAMES_SIZE);
}
```

The errors that I am getting is as below:
`Non-zero status code returned while running Reshape node. Name:'BoxPredictor_0/Reshape' Status Message: D:\3\s\onnxruntime\core\providers\dml\DmlExecutionProvider\src\MLOperatorAuthorImpl.cpp(1638)\onnxruntime.dll!00007FFA2A61671D: (caller: 00007FFA2A98091A) Exception(2) tid(7428) 8007023E {Application Error}`

However, I believe the problem is due to DML. This is because once I commented `OrtSessionOptionsAppendExecutionProvider_DML(_session_options, 0);` in my code, I am able to run the gstreamer pipeline successfully with inference running on CPU.

In short, I am able to run the model with a standalone sample app but failed when I integrate it into a gstreamer plugin.

FYI, below is my system information:
Windows 10 Build 18363
Intel i7-8665U + UHD 620

[ONNX-DirectML Nuget Package v1.7.0](https://www.nuget.org/packages/Microsoft.ML.OnnxRuntime.DirectML)

Thanks,
Vincent

**Comment by vincent911001** [comment]
Hi, 

I have followed the guide in [https://github.com/microsoft/onnxruntime/issues/3345#issuecomment-713870288](https://github.com/microsoft/onnxruntime/issues/3345#issuecomment-713870288) to force d3d12 debug message output to DebugView and got the following outputs:

```
[12764] DXGI WARNING: IDXGIFactory::CreateSwapChain/IDXGISwapChain::ResizeBuffers: The buffer height inferred from the output window is zero. Taking 8 as a reasonable default instead [ MISCELLANEOUS WARNING #2: ]
[12764] DXGI WARNING: IDXGIFactory::CreateSwapChain/IDXGISwapChain::ResizeBuffers: The buffer height inferred from the output window is zero. Taking 8 as a reasonable default instead [ MISCELLANEOUS WARNING #2: ]
[12764] D3D12 MESSAGE: ID3D12GraphicsCommandList::CreateMetaCommand: MetaCommand parameters are not supported by the current system configuration. [ STATE_CREATION MESSAGE #1172: META_COMMAND_UNSUPPORTED_PARAMS]
[12764] D3D12 MESSAGE: ID3D12GraphicsCommandList::CreateMetaCommand: MetaCommand parameters are not supported by the current system configuration. [ STATE_CREATION MESSAGE #1172: META_COMMAND_UNSUPPORTED_PARAMS]
[12764] D3D12 MESSAGE: ID3D12GraphicsCommandList::CreateMetaCommand: MetaCommand parameters are not supported by the current system configuration. [ STATE_CREATION MESSAGE #1172: META_COMMAND_UNSUPPORTED_PARAMS]
[12764] D3D12 MESSAGE: ID3D12GraphicsCommandList::CreateMetaCommand: MetaCommand parameters are not supported by the current system configuration. [ STATE_CREATION MESSAGE #1172: META_COMMAND_UNSUPPORTED_PARAMS]
[12764] D3D12 MESSAGE: ID3D12GraphicsCommandList::CreateMetaCommand: MetaCommand parameters are not supported by the current system configuration. [ STATE_CREATION MESSAGE #1172: META_COMMAND_UNSUPPORTED_PARAMS]
[12764] D3D12 MESSAGE: ID3D12GraphicsCommandList::CreateMetaCommand: MetaCommand parameters are not supported by the current system configuration. [ STATE_CREATION MESSAGE #1172: META_COMMAND_UNSUPPORTED_PARAMS]
[12764] D3D12 MESSAGE: ID3D12GraphicsCommandList::CreateMetaCommand: MetaCommand parameters are not supported by the current system configuration. [ STATE_CREATION MESSAGE #1172: META_COMMAND_UNSUPPORTED_PARAMS]
[12764] D3D12 MESSAGE: ID3D12GraphicsCommandList::CreateMetaCommand: MetaCommand parameters are not supported by the current system configuration. [ STATE_CREATION MESSAGE #1172: META_COMMAND_UNSUPPORTED_PARAMS]
[12764] D3D12 MESSAGE: ID3D12GraphicsCommandList::CreateMetaCommand: MetaCommand parameters are not supported by the current system configuration. [ STATE_CREATION MESSAGE #1172: META_COMMAND_UNSUPPORTED_PARAMS]

```

![dml_error_trace](https://user-images.githubusercontent.com/12131623/116505717-38967780-a8ee-11eb-8e1a-8aa2bfd12d49.PNG)

Please advise.

Thanks,
Vincent

**Comment by vincent911001** [comment]
Hi,

Found out that the model is unable to taking in high resolution image, have to size the image down before submitting it as input tensor.

Thanks,
Vincent
